common/tasks/interleaved_weighted_round_robin_test.go

```diff
@@ -437,6 +437,10 @@ func (s *interleavedWeightedRoundRobinSchedulerSuite) TestDeleteInactiveChannels
 
 	s.ts.Advance(30 * time.Minute)
 
+	// Sleeping for a small duration for the current event loop to finish.
+	// We read ts.Now() before the loop begins and reuse it in the loop.
+	time.Sleep(100 * time.Millisecond) //nolint:forbidigo
```

Comment:
How difficult would it be to wire the fake clock all the way to the dispatch loop?

Reply:
We have the time source inserted to the dispatch loop. But it may read the old value and enter the loop. If this happens, LastUpdatedTime will be the old time for all tasks processed in the loop.

Reply:
is the goal here to make sure the dispatch loop can see the advanced time from the eventTimeSource? If so, use a s.Eventually to wait for Now() to be advanced before submitting any new tasks?

Reply:
That was not the issue.

Consider that a thread has executed this line: https://github.com/temporalio/temporal/blob/7bfdbffa14cee01be54d9daeb8cdca073a71ffb9/common/tasks/interleaved_weighted_round_robin.go#L370-L370 and entered `LoopDispatch` below that. 

At this exact time, we might update time source.   ts.Now() is updated. We might add new tasks now. But the above thread has read the old time value already. These new task's LastActiveTime will be updated to the old time value that is already read.

Because of this, these channels have an old LastUpdatedTime and it will be cleaned up when we advance time in test. One another way to fix this is to run ts.Now() on line #375 for each task.

Reply:
Ah I see. Thanks for the explanation. 
yeah that's pretty hard to get rid of. Maybe add more context to the comment, essentially saying we have no way to know if a dispatch loop finished or not.